src/main/java/com/wolox/training/controllers/UserController.java

```diff
@@ -115,4 +116,20 @@ public User removeBookToUser(@ApiParam(name = "userId", required = true)@PathVar
         user.removeBook(book);
         return userRepository.save(user);
     }
+
+    @PutMapping("/{id}/newPassword")
+    public User newPassword(@PathVariable Long id, @RequestBody Map<String, String> parameters) throws UserNotFoundException, OldPasswordMismatchEception {
```

Comment:
In the real life, you should also ask the username, you'd be also making a double check by controlling that whoever is trying to update the password, also knows the username